projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,151 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  options: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+  linkedResource: ReadResource;
+  linkedResourceIRI: string;
+
+  valueChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, @Inject(KnoraApiConnectionToken) private knoraApiConnection: KnoraApiConnection) {
+    super();
+  }
+
+  /**
+   * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+   * this is to be used for update and new linked resources
+   *
+   * @param label to be searched
+   */
+    searchByLabel(searchTerm: string): ReadResource[] {
+      this.restrictToResourceClass = this.displayValue.linkedResource.type;
```

Comment:
Could you make this a local variable?

What about the value creation case? I think you need to look at the ontology for the property in question to find about about the restriction.

Reply:
Probably the component needs an additional input where the property definition can be passed in.

Reply:
@tobiasschweizer yes, that is right. How can I check the object type of a link property from Ontology? Is there a function in @knora/Api for returning property definition?

Reply:
You can get it from the `entityDefinition` on the resource that you are editing / creating a value for. I think you need an additional Input for your component where you pass this information in. I can have a look and make an example if this helps.

Reply:
Here is the member on `ReadResource`:

https://github.com/dasch-swiss/knora-api-js-lib/blob/03e74aec56d7d005b866e6f48ddf6b771fa280a0/src/models/v2/resources/read/read-resource.ts#L45

From that collection, the information for the property in question would have to be returned. 

That would be https://github.com/dasch-swiss/knora-api-js-lib/blob/03e74aec56d7d005b866e6f48ddf6b771fa280a0/src/models/v2/ontologies/property-definition.ts#L9

The `objectType` is optional. So when absent, the restriction is a `knora-api:Resource`

Reply:
for the edit case, you can get it from here

https://github.com/dasch-swiss/knora-ui-ng-lib/blob/e9cae9fa155c22a02385dc3a6558243525d7b3a1/projects/knora-ui/src/lib/viewer/operations/display-edit/display-edit.component.ts#L28

and pass it along in the template.

Reply:
Ok, I got the property definition from ontology and passed it as input to the component.
passing the parameter as below in displayEdit
` <kui-link-value #displayVal *ngSwitchCase="constants.LinkValue" [mode]="mode" [displayValue]="displayValue" [propertyDesc]="parentResource.entityInfo"></kui-link-value>
    `
and reading it as input in my component
`  @Input() propertyDesc?: IResourceClassAndPropertyDefinitions;
console.log(this.propertyDesc.properties[this.displayValue.property])
`
See the output of the console command given above here:
![image](https://user-images.githubusercontent.com/8567642/76531043-e5a4fb00-6474-11ea-86b5-02ae855f866f.png)

Keeping in mind how we define the link properties in ontology: we specify the type of the object resource in property definition itself e.g. "hasOtherThing" but the "hasOtherThingValue" is actually the LinkValue. In app.component, I read "hasOtherThingValue" to set the value of the link. (i.e. this.displayValue.property is http://0.0.0.0:3333/ontology/0001/anything/v2#hasOtherThingValue)

The question is how to read the objectType of the "hasOtherThing"? Regex is not an option, one cannot simply trim "Value" from the end of this.displayValue.property.

I believe there should be a way to get the linking property ("hasOtherThing") definition from the definition of its linkValue ("hasOtherThingValue").

Reply:
> I believe there should be a way to get the linking property ("hasOtherThing") definition from the definition of its linkValue ("hasOtherThingValue").

True, good point! I could make a function in knora-api-js-lib. 
In Knora, there is the same logic.

Reply:
Ok, I will try to do this and let you know!

Reply:
Please check https://github.com/dasch-swiss/knora-api-js-lib/pull/163